src/eventbridge/transform.ts

```diff
@@ -0,0 +1,61 @@
+import { aws_events } from "aws-cdk-lib";
+import { FunctionDecl } from "../declaration";
+import { synthesizeEventBridgeTargets } from "./targets";
+import { Function } from "../function";
+import { LambdaTargetProps, pipe } from "./target";
+import { EventBusRuleInput } from "./types";
+import { EventBusRule } from "./rule";
+
+/**
+ * A function interface used by the {@link EventBusRule}'s map function.
+ *
+ * event is the event matched by the rule. This argument is optional.
+ * $utils is a collection of built-in utilities wrapping EventBridge TargetInputs like contextual constants avaliable to the transformer.
+ */
+export type EventTransformFunction<E extends EventBusRuleInput, O = any> = (
+  event: E,
+  $utils: EventTransformUtils
+) => O;
+
+/**
+ * https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-transform-target-input.html#eb-transform-input-predefined
+ */
+export interface EventTransformUtils {
+  context: {
+    ruleArn: string;
+    ruleName: string;
+    ingestionTime: string;
+    eventJson: string;
+  };
+}
+
+/**
+ * Represents an event that has been transformed using Target Input Transformers.
+ * https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-transform-target-input.html
```

Comment:
use `@see`

Reply:
Nice, will keep this handy for more rich docs.

https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#see